coreneuron/io/reports/report_handler.cpp

```diff
@@ -0,0 +1,230 @@
+
+namespace coreneuron {
+
+void ReportHandler::create_report(double dt, double tstop, double delay) {
+#if defined(ENABLE_REPORTINGLIB) || defined(ENABLE_SONATA_REPORTS)
+    if (m_report_config.start < t) {
+        m_report_config.start = t;
+    }
+    m_report_config.stop = std::min(m_report_config.stop, tstop);
+
+    m_report_config.mech_id = nrn_get_mechtype(m_report_config.mech_name);
+    if (m_report_config.type == SynapseReport && m_report_config.mech_id == -1) {
+        std::cerr << "[ERROR] mechanism to report: " << m_report_config.mech_name
+                  << " is not mapped in this simulation, cannot report on it \n";
+        nrn_abort(1);
+    }
+    for (int ith = 0; ith < nrn_nthread; ++ith) {
+        NrnThread& nt = nrn_threads[ith];
+        if (!nt.ncell) {
+            continue;
+        }
+        const std::vector<int>& nodes_to_gid = map_gids(nt);
+        VarsToReport vars_to_report;
+        switch (m_report_config.type) {
+            case SomaReport:
+                vars_to_report = get_soma_vars_to_report(nt, m_report_config.target, nt._actual_v);
+                register_soma_report(nt, m_report_config, vars_to_report);
+                break;
+            case CompartmentReport:
+                vars_to_report =
+                    get_compartment_vars_to_report(nt, m_report_config.target, nt._actual_v);
+                register_compartment_report(nt, m_report_config, vars_to_report);
+                break;
+            case IMembraneReport:
+                vars_to_report = get_compartment_vars_to_report(nt, m_report_config.target,
+                                                                nt.nrn_fast_imem->nrn_sav_rhs);
+                register_compartment_report(nt, m_report_config, vars_to_report);
+                break;
+            default:
+                vars_to_report = get_custom_vars_to_report(nt, m_report_config, nodes_to_gid);
+                register_custom_report(nt, m_report_config, vars_to_report);
+        }
+        if (!vars_to_report.empty()) {
+            std::unique_ptr<ReportEvent> report_event =
+                std::make_unique<ReportEvent>(dt, t, vars_to_report, m_report_config.output_path);
+            report_event->send(t, net_cvode_instance, &nt);
+            m_report_events.push_back(std::move(report_event));
+        }
+    }
+#else
+    if (nrnmpi_myid == 0) {
+        std::cerr << "[WARNING] : Can't enable reports, recompile with ReportingLib! \n";
+    }
+#endif  // defined(ENABLE_REPORTINGLIB) || defined(ENABLE_SONATA_REPORTS)
+}
+
+#if defined(ENABLE_REPORTINGLIB) || defined(ENABLE_SONATA_REPORTS)
+void ReportHandler::register_soma_report(const NrnThread& nt,
+                                         ReportConfiguration& config,
+                                         const VarsToReport& vars_to_report) {
+    if (nrnmpi_myid == 0) {
+        std::cerr << "[WARNING] : Report '" << config.output_path << "' not supported!\n";
```

Comment:
I don't know what it means when `nrnmpi_myid == 0`, but the error message does not provide much more information. It should be improved if there is a more precise reason.

Besides if the exclamation mark at the end of the logs messages is uncommon in CoreNeuron, then IMO they should be removed.

Reply:
Same question and I couldn't figure out  why all reports here print the warning with `config.output_path` ?

Reply:
This is called only when the format of the report doesn't exist. Otherwise the subclasses sonata_report_handler or binary_report_handler are called.